app/src/main/java/org/schabi/newpipe/util/SaveUploaderUrlHelper.java

```diff
@@ -0,0 +1,78 @@
+package org.schabi.newpipe.util;
+
+import static org.schabi.newpipe.extractor.utils.Utils.isNullOrEmpty;
+
+import android.content.Context;
+import android.widget.Toast;
+
+import androidx.fragment.app.Fragment;
+
+import org.schabi.newpipe.NewPipeDatabase;
+import org.schabi.newpipe.R;
+import org.schabi.newpipe.error.ErrorInfo;
+import org.schabi.newpipe.error.ErrorUtil;
+import org.schabi.newpipe.error.UserAction;
+import org.schabi.newpipe.extractor.stream.StreamInfoItem;
+import org.schabi.newpipe.player.playqueue.PlayQueueItem;
+
+import io.reactivex.rxjava3.android.schedulers.AndroidSchedulers;
+import io.reactivex.rxjava3.schedulers.Schedulers;
+
+/**
+ * Utility class for putting the uploader url into the database - when required.
+ */
+public final class SaveUploaderUrlHelper {
+    private SaveUploaderUrlHelper() {
+    }
+
+    // Public functions which call the function that does
+    // the actual work with the correct parameters
+    public static void saveUploaderUrlIfNeeded(final Fragment fragment,
+                                               final StreamInfoItem infoItem,
+                                               final SaveUploaderUrlCallback callback) {
+        saveUploaderUrlIfNeeded(fragment.requireContext(),
+                infoItem.getServiceId(),
+                infoItem.getUrl(),
+                infoItem.getUploaderUrl(),
+                callback);
+    }
+    public static void saveUploaderUrlIfNeeded(final Context context,
+                                               final PlayQueueItem queueItem,
+                                               final SaveUploaderUrlCallback callback) {
+        saveUploaderUrlIfNeeded(context,
+                queueItem.getServiceId(),
+                queueItem.getUrl(),
+                queueItem.getUploaderUrl(),
+                callback);
+    }
+
+    private static void saveUploaderUrlIfNeeded(final Context context,
```

Comment:
Also add a javadoc here, explaining that this function loads fetches the uploaderUrl if the item does not already have it, and in that case saves it to the db. Anyway the callback is always called with a valid channel url.

Reply:
Ok, I'll do that.

> Anyway the callback is always called with a valid channel url.

What does that mean?

Reply:
I mean, in the javadoc explain that the callback is always called with a valid uploader url (unless an error has occurred). From the name of the function it is unclear that the function fetches the uploader url if it is not available.

Reply:
Done in https://github.com/TeamNewPipe/NewPipe/commit/973fd8f6baf1beff4dcf6525205de2bd933138b7